tutorials/RandomNumberGeneration/ReferenceImplementation.qs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+//////////////////////////////////////////////////////////////////////
+// This file contains reference solutions to all tasks.
+// You should not modify anything in this file.
+// We recommend that you try to solve the tasks yourself first,
+// but feel free to look up the solution if you get stuck.
+//////////////////////////////////////////////////////////////////////
+
+namespace Quantum.Kata.RandomNumberGeneration {
+    
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Measurement;
+    open Microsoft.Quantum.Math;
+    
+
+    // Exercise 1.
+    operation RandomBit_Reference () : Int {
+        using (q = Qubit()) {
+            H(q);
+            return MResetZ(q) == Zero ? 0 | 1;
+        }
+    }
+
+    // Exercise 2. 
+    operation RandomTwoBits_Reference () : Int {
+        let firstBit = RandomBit_Reference();
```

Comment:
I would simplify it to `return RandomBit_Reference() * 2 + RandomBit_Reference();`